index.html

```diff
@@ -269,6 +258,55 @@ <h4>
             or bypassing the UI if there is only a single share target.
           </div>
         </section>
+        <section>
+          <h3>
+            Validate share data
+          </h3>
+          <p>
+            To <dfn>validate share data</dfn> with |data:ShareData| and
+            |base:URL|, run the following steps:
+          </p>
+          <ol class="algorithm">
+            <li>If none of |data|'s members {{ShareData/title}},
+            {{ShareData/text}}, or {{ShareData/url}} or {{ShareData/files}} are
+            present, return false.
+            </li>
+            <li>Let |titleTextOrUrl:boolean| be true if any of
+            {{ShareData/title}}, or {{ShareData/text}}, or {{ShareData/url}} is
+            present.
+            </li>
+            <li>If |data|'s {{ShareData/files}} member is present:
+              <ol>
+                <li>If |titleTextOrUrl| is false, and if |data|'s
+                {{ShareData/files}} member is empty, or return false.
+                  <p class="note">
+                    This causes a `{ files: [] }` dictionary to be treated as
+                    an empty dictionary. However, passing a dictionary like
+                    `{text: "text" files: []}` is fine, as `files` is just
+                    ignored.
+                  </p>
+                </li>
+                <li>If the implementation does not support file sharing, return
+                false.
+                </li>
```

Comment:
Is this the point where you want to add validation for file content type/extension?

Reply:
yes, this would be the place.... open question:

 1. Should we require both the filename and the type? Or should be require just one or the other. I'm leaning towards only requiring the filename, which must include an "allowed extension". If the type is included, it must be one that maps to the type (i.e., ".jpg" maps to "image/jpg" or "image/jpeg"). 
 1. Should we allow dot files (e.g,., ".png")? My feeling is no because of how they behave in some OSs.

Reply:
An issue discussion is probably a better place to discuss that.  Should we use an issue?

Reply:
Yes, let's pick this up in https://github.com/w3c/web-share/issues/127